src/Microsoft.Extensions.Primitives/ChangeToken.cs

```diff
@@ -36,7 +36,12 @@ public static IDisposable OnChange(Func<IChangeToken> changeTokenProducer, Actio
                 // If the token changes after we take the token, then we'll process the update immediately upon
                 // registering the callback.
                 var t = changeTokenProducer();
-                changeTokenConsumer();
+                try
+                {
+                    changeTokenConsumer();
+                }
+                catch // Uncaught exceptions are really bad here
+                { }
```

Comment:
Regarding using try/catch vs. try/finally: Do we understand the full consequences of swallowing the exception? This code usually gets executed from a different thread in a "fire and forget" fashion, e.g. triggered by a file system change in which case I believe the exception wouldn't be harmful if we had try/finally , but I think it can also be invoked programmatically in the same thread, e.g. for configuration, calling `ConfigurationReloadToken.OnReload()`, in which case it might be weird for the exception to just disappear.